src/lib.rs

```diff
@@ -82,22 +77,36 @@ impl RustSemsimian {
         subject_terms: &HashSet<TermID>,
         object_terms: &HashSet<TermID>,
         predicates: &Option<HashSet<Predicate>>,
-    ) -> HashMap<TermID, HashMap<TermID, (f64, f64)>> {
+    ) -> HashMap<TermID, HashMap<TermID, (f64, f64, f64, HashSet<TermID>)>> {
         let self_shared = Arc::new(Mutex::new(self.clone()));
 
-        let similarity_map: HashMap<TermID, HashMap<TermID, (f64, f64)>> = subject_terms
-            .par_iter() // parallelize computations
-            .map(|subject| {
-                let mut subject_similarities: HashMap<TermID, (f64, f64)> = HashMap::new();
-                for object in object_terms.iter() {
-                    let mut self_locked = self_shared.lock().unwrap();
-                    let jaccard_sim = self_locked.jaccard_similarity(subject, object, predicates);
-                    let resnik_sim = self_locked.resnik_similarity(subject, object, predicates);
-                    subject_similarities.insert(object.clone(), (resnik_sim, jaccard_sim));
-                }
-                (subject.clone(), subject_similarities)
-            })
-            .collect();
+        let similarity_map: HashMap<TermID, HashMap<TermID, (f64, f64, f64, HashSet<TermID>)>> =
```

Comment:
having these three similarity values side by side without any labels is probably eventually going to cause problems - is there a way to explicitly say which is Jaccard, Resnik and Phenodigm in the data structure? (Also I realize you probably didn't design this struct)

Reply:
Could be a `HashMap<String, String>` which many get complex
```
similarity_map: HashMap<TermID, HashMap<TermID, HashSet<HashMap<String, f64>, HashMap<String, f64>,HashMap<String,  f64>, HashSet<TermID>)>>
```

My eyes are already squinting ... lol . 

Alternatively, it could be a new struct
```
similarity_map: HashMap<TermID, HashMap<TermID, (Jaccard, Resnik, Phenodigm, HashSet<TermID>)>
```
Just like `TermID = String` the others are `f64`

Reply:
I'd vote for these being specific types if possible

Reply:
Done